src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ServiceUtilHelper.cs

```diff
@@ -43,38 +63,39 @@ public static bool TryEnsureRootCertificateInstalled()
                     // for the root certificate and install it locally if it
                     // is not already in the store.
                     rootCertificate = InstallRootCertificateFromServer();
-                }
-                catch (Exception ex)
-                {
-                    System.Console.WriteLine(String.Format("Attempt to install root certificate failed:{0}{1}", 
-                                                            Environment.NewLine, ex.ToString()));
-                }
-
-                // If we had a certificate from the service endpoint, verify it was installed
-                // by retrieving it from the store by thumbprint.
-                if (rootCertificate != null)
-                {
-                    thumbprint = rootCertificate.Thumbprint;
-                    rootCertificate = CertificateManager.RootCertificateFromThumprint(thumbprint);
-                }
 
-                if (rootCertificate != null)
-                {
-                    System.Console.WriteLine(String.Format("Using root certificate:{0}{1}",
-                                            Environment.NewLine, rootCertificate));
+                    // If we had a certificate from the service endpoint, verify it was installed
+                    // by retrieving it from the store by thumbprint.
+                    if (rootCertificate != null)
+                    {
+                        thumbprint = rootCertificate.Thumbprint;
+                        rootCertificate = CertificateManager.RootCertificateFromThumprint(thumbprint);
+                    }
+
+                    if (rootCertificate != null)
```

Comment:
CertificateManager.RootCertificateFromThumbprint can potentially return null.  I'll move the second 'if' inside the first, but it still needs to check if it was able to obtain the cert from the store by thumbprint after it obtained it from the server and believes it has installed it.  Actually, @zhenlan -- this second check dates back to your requiring we verify the cert really was in the store :grin: